caf_solution/add-ons/azure_devops_v1/azdo_service_endpoints_acr.tf

```diff
@@ -0,0 +1,26 @@
+data "azurerm_container_registry" "acr" {
+  for_each            = var.service_endpoints_acr
+  name                = local.remote.azure_container_registries[each.value.azure_container_registry.lz_key][each.value.azure_container_registry.key].name
+  resource_group_name = local.remote.azure_container_registries[each.value.azure_container_registry.lz_key][each.value.azure_container_registry.key].resource_group_name
+}
+
+resource "azuredevops_serviceendpoint_dockerregistry" "acr" {
+  for_each = var.service_endpoints_acr
+
+  project_id            = data.azuredevops_project.project[each.value.project_key].id
+  service_endpoint_name = each.value.endpoint_name
+
+  docker_registry = format("https://%s", data.azurerm_container_registry.acr[each.key].login_server)
+  docker_username = data.azurerm_container_registry.acr[each.key].admin_username
```

Comment:
Note that unfortunately this will only work if the admin account is enabled on the ACR, which often will not be the case.

One approach I've been using to access ACR via a pipeline is to use the MSI that's associated with the build agent. Once the MSI has been granted AcrPull/AcrPush on the registry you can simply do `az login --identity` followed by `az acr login --name ...`. The only issue is getting the value for `--name`. In my case I've been using the AZDO variable groups with remote object lookups for this, see https://github.com/Azure/caf-terraform-landingzones/pull/290 for an example.

I think both approaches are valid, would like to hear others thoughts here.